src/mk/tools/just.py

```diff
@@ -0,0 +1,50 @@
+from __future__ import annotations
+
+import os
+import re
+from pathlib import Path
+
+from mk.exec import run_or_fail
+from mk.tools import Action, Tool
+
+
+class JustTool(Tool):
+    name = "just"
+
+    def __init__(self) -> None:
+        super().__init__(self)
+        self.justfile: str | None = None
+
+    def run(self, action: Action | None = None) -> None:
+        cmd = ["just"]
+        if action:
+            cmd.append(action.name)
+        run_or_fail(cmd, tee=True)
+
+    def is_present(self, path: Path) -> bool:
+        for name in ["Justfile", "justfile", "JUSTFILE", ".justfile"]:
+            justfile = os.path.join(path, name)
+            if os.path.isfile(justfile):
+                self.justfile = justfile
+                return True
+        return False
+
+    def actions(self) -> list[Action]:
+        actions = []
+        if not self.justfile:
+            msg = "Justfile not found"
+            raise RuntimeError(msg)
+        # pylint: disable=duplicate-code
+        with open(self.justfile, encoding="utf-8") as file:
+            for line in file.readlines():
```

Comment:
It's more efficient to iterate over the file directly

```suggestion
            for line in file:
```